### DISCUSSION TITLE
Technical details about Oauth tokens

### DISCUSSION THREAD

**Discussion by itsamemarkus** [iterate-ch/cyberduck#15359] [Q&A] [answered]
If I understand this correctly oauth tokens (for example for OneDrive) are generated for the requesting client and not for some Cyberduck cloud middleware, right?

I think it would really help if you could explain what's happening on a technical level in the background and if iterate has ever access to the authorization code or if it's just the client. Maybe this would be a good place for it:

https://docs.cyberduck.io/protocols/onedrive/

How permissions look on onedrive:

<img width="1122" alt="image" src="https://github.com/iterate-ch/cyberduck/assets/34944988/295663cb-78fd-4497-bae1-52347916b6eb">

**Comment by ylangisc** [comment, marked as answer]
There is no middleware or proxy in between. The auth process is just between Mountain Duck and the identity service provider (IdP) of your cloud storage provider.
